napari/utils/misc.py

```diff
@@ -477,3 +477,35 @@ def dir_hash(path: Union[str, Path], include_paths=True, ignore_hidden=True):
                 _hash.update(''.join(fparts).encode())
 
     return _hash.hexdigest()
+
+
+def combine_signatures(
+    *objects: Callable, return_annotation=inspect.Signature.empty, exclude=()
+) -> inspect.Signature:
+    """Create combined Signature from objects, excluding names in `exclude`.
+
+    Parameters
+    ----------
+    *objects : Callable
+        callables whose signatures should be combined
+    return_annotation : [type], optional
+        The return annotation to use for combined signature, by default
+        inspect.Signature.empty (as it's ambiguous)
+    exclude : tuple, optional
+        Parameter names to exclude from the combined signature (such as
+        'self'), by default ()
+
+    Returns
+    -------
+    inspect.Signature
+        Signature object with the combined signature. Reminder, str(signature)
+        provides a very nice repr for code generation.
+    """
+    params = itertools.chain(
+        *(inspect.signature(o).parameters.values() for o in objects)
+    )
+    new_params = sorted(
+        (p for p in params if p.name not in exclude),
+        key=lambda p: p.kind,
+    )
```

Comment:
[`Parameter.kind`](https://docs.python.org/3/library/inspect.html#inspect.Parameter.kind) is things like `KEYWORD_ONLY` and stuff... so if you don't sort on that, you can actually get errors.  However, it won't have an effect on the order otherwise (they will stay in object order unless it would be an error to do so)